src/main/java/cinema/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package cinema.dao.impl;
+
+import cinema.dao.OrderDao;
+import cinema.exception.DataProcessingException;
+import cinema.lib.Dao;
+import cinema.model.Order;
+import cinema.model.User;
+import cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+    @Override
+    public Order add(Order order) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(order);
+            transaction.commit();
+            return order;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Couldn't insert order " + order, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Order> getOrdersByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Order> orders = session.createQuery("FROM Order o "
+                    + "LEFT JOIN FETCH o.tickets "
+                    + "WHERE o.user = :user", Order.class);
```

Comment:
let's use distinct keyword in order to get rid of duplications if our order has more than one ticket

Reply:
done